### PR TITLE
Add `clean_text_or_reply` util.

### LINKED ISSUE #100: Add a `convert_to_cleaned_text_or_reply` util
As per python-discord/sir-lancebot#1074, we should add a `CleanedTextOrReply` converter which will allow an argument to be provided either by text, or by replying to a message (if an invocation has both then use the text).

There's currently no file for converters, so we'd need a new file -- either `botcore/converters.py` as per bot structure or `botcore/utils/converters.py` as per sir-lancebot structure. 

I'd like to implement this myself.

### PR DESCRIPTION
**Pull request by TizzySaurus**
Closes #100.

Does what the title says.

This util function will clean `text` if provided, else try and clean the referenced message, or if `text` isn't provided and was unable to get a reference message will ~~return `None`~~ raise a BadArgument error, as per [Scaleois' review](https://github.com/python-discord/bot-core/pull/101#discussion_r928096299). 

This is useful for commands such as lancebot's `.uwu` and `.randomcase` where users can either provide a text argument, or reply to a message and have the bot use the replied message's content.

~~NB: Currently a draft as untested~~ Has now been tested so ready for review

### CODE REVIEW

**Comment by TizzySaurus** [comment]
This has now been tested and it works :+1:

Sample code usage:
image


Result of said sample:
image


This is now ready for review :smile:

File: botcore/utils/commands.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        return await clean_content_converter.convert(ctx, text)
+
+    if (
+        (replied_message := getattr(ctx.message.reference, "resolved", None))  # message has a cached reference
+        and isinstance(replied_message, Message)  # referenced message hasn't been deleted
+    ):
+        return await clean_content_converter.convert(ctx, ctx.message.reference.resolved.content)
+
+    # No text provided, and either no message was referenced or we can't access the content
+    return None
```
Comment (HassanAbouelela) [inline botcore/utils/commands.py:21]: Should we return None here? If we throw an error, it saves having to include validation logic in the caller. If the caller wants to do non-standard behavior, they can catch the exception.

File: botcore/utils/commands.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from typing import Optional
+
+from discord import Message
+from discord.ext.commands import Context, clean_content
+
+
+async def clean_text_or_reply(ctx: Context, text: Optional[str] = None) -> Optional[str]:
+    """Returns cleaned version of `text`, if given, else referenced message, if found, else `None`."""
```
Comment (HassanAbouelela) [inline botcore/utils/commands.py:8]: Please include a description for the paramaters and return in the docstring. If you aren't familiar, refer to the readme section on [writing docstrings](https://github.com/python-discord/bot-core/blob/main/docs/README.md#docstrings)

**Review by HassanAbouelela — CHANGES_REQUESTED** [review]

File: botcore/utils/commands.py (hunk 1 above)
Comment (TizzySaurus) [reply botcore/utils/commands.py:21]: Agree, done :+1:

File: botcore/utils/commands.py (hunk 2 above)
Comment (TizzySaurus) [reply botcore/utils/commands.py:8]: :+1:

**Review by HassanAbouelela — APPROVED** [review]

**Pull request merged**